app/components/progressive_image/progressive_image.test.js

```diff
@@ -22,22 +22,34 @@ describe('ProgressiveImage', () => {
         tintDefaultSource: false,
     };
 
-    test('should match snapshot when just a image loads', () => {
-        const wrapper = shallow(
-            <ProgressiveImage {...baseProps}/>
-        );
+    test('should match snapshot', () => {
+        const wrapper = shallowWithIntl(<ProgressiveImage {...baseProps}/>);
         expect(wrapper.getElement()).toMatchSnapshot();
     });
 
-    test('should match snapshot when no thumbnail', () => {
-        const props = {
-            ...baseProps,
-            thumbnailUri: null,
-        };
+    test('should load image', () => {
+        const wrapper = shallowWithIntl(<ProgressiveImage {...baseProps}/>);
+        const instance = wrapper.instance();
+        const load = jest.spyOn(instance, 'load');
+        load();
+        expect(load).toHaveBeenCalled();
+    });
 
-        const wrapper = shallow(
-            <ProgressiveImage {...props}/>
-        );
-        expect(wrapper.getElement()).toMatchSnapshot();
+    test('should set image state', () => {
+        const wrapper = shallowWithIntl(<ProgressiveImage {...baseProps}/>);
+        const instance = wrapper.instance();
+        const setImage = jest.spyOn(instance, 'setImage');
+        setImage('test.png');
+        expect(setImage).toHaveBeenCalled();
```

Comment:
same as above, but in this case I don't think you need to check that it was called.

In this test you should check that `setImage` is called after ImageCacheManager.cache executes the callback.

Reply:
Can be,
```javascript
jest.useFakeTimers();

describe('ProgressiveImage', () => {
...
    test('should set image when imageUri is set but not the thumbnailUri', () => {
        const wrapper = shallow(
            <ProgressiveImage
                {...baseProps}
                thumbnailUri={null}
            />
        );
        const instance = wrapper.instance();
        jest.spyOn(instance, 'setImage');

        instance.componentDidMount();
        jest.runAllTimers();
        expect(instance.setImage).toHaveBeenCalledTimes(1);
    });
```